pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/NonAggregationGroupByToDistinctQueryRewriter.java

```diff
@@ -41,7 +40,7 @@
  * SELECT col1 AS c1 FROM foo GROUP BY col1 --> SELECT DISTINCT col1 AS c1 FROM foo
  * SELECT col1, col1 AS c1, col2 FROM foo GROUP BY col1, col2 --> SELECT DISTINCT col1, col1 AS ci, col2 FROM foo
  *
- * Unsupported queries:
+ * Not rewritten queries:
```

Comment:
@tarun11Mavani : Are we sure we are only allowing the valid queries to pass through here?

Invalid queries such as the following should continue to be disallowed. `SELECT concat(col1, someOtherCol) from table GROUP BY col1, col2`.

cc: @vvivekiyer

Reply:
The idea is to rewrite only the queries that can actually be rewritten. The validation part happens right after all the rewrites are done. I think rewriters should just focus on rewriting the query when it makes sense, without worrying about validating it.

As mentioned above, these type of query will fail due to a validation process that occurs immediately after all rewrites are completed in the [CalciteSqlParser](https://github.com/apache/pinot/blob/53fbf88027c47b3c6f7d1526576ba0bd257fe9d5/pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java#L550). Specifically, the [validateGroupByClause](https://github.com/apache/pinot/blob/53fbf88027c47b3c6f7d1526576ba0bd257fe9d5/pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java#L550) function is invoked to recursively check whether an expression contains any references that are not included in the GROUP BY clause.

Reply:
Can you update the error in validateGroupByClause. Right now it says "should appear in GROUP BY clause". We can instead say: "should be functionally dependent on the columns used in group-by":

Reply:
Good suggestion. Update the error message!!